### ISSUE TITLE
Is this a bug?

### ISSUE THREAD

**Issue by answer-miao** [M-3LAB/Look-Inside-for-More#7]
thanks for your contribution!!!!!    in the main.py

torch.cuda.empty_cache()
        PatchCore2.set_deep_feature_extractor()
        _ = PatchCore2.fit_with_limit_size_pmae(train_loader, memory_size)
        aggregator_fpfh = {"scores": [], "segmentations": []}
        scores_fpfh, segmentations_fpfh, _, _ = PatchCore2.predict_pmae(
            test_loader
        )
        aggregator_fpfh["scores"].append(scores_xyz)
        scores_fpfh = np.array(aggregator_fpfh["scores"])
        min_scores_fpfh = scores_fpfh.min(axis=-1).reshape(-1, 1)
        max_scores_fpfh = scores_fpfh.max(axis=-1).reshape(-1, 1)
        scores_fpfh = (scores_fpfh - min_scores_fpfh) / (max_scores_fpfh - min_scores_fpfh)
        scores_fpfh = np.mean(scores_fpfh, axis=0)
        
        del PatchCore2

the "aggregator_fpfh["scores"].append(scores_xyz)" should be "aggregator_fpfh["scores"].append(scores_fpfh)"????

**Comment by hzzzzzhappy** [comment]
Hi,

This is derived from the original code structure of reg3d-ad. We employ coordinate splitting combined with feature splitting, consistent with the original reg3d-ad architecture. FPFH is merely a list designation.

**Comment by answer-miao** [comment]
thanks for your reply!
But here, scores_fpfh is immediately overwritten as soon as it is output. Is this correct?



torch.cuda.empty_cache()
PatchCore2.set_deep_feature_extractor()
_ = PatchCore2.fit_with_limit_size_pmae(train_loader, memory_size)
aggregator_fpfh = {"scores": [], "segmentations": []}
scores_fpfh, segmentations_fpfh, _, _ = PatchCore2.predict_pmae(
test_loader
)
aggregator_fpfh["scores"].append(scores_xyz)
scores_fpfh = np.array(aggregator_fpfh["scores"])
min_scores_fpfh = scores_fpfh.min(axis=-1).reshape(-1, 1)
max_scores_fpfh = scores_fpfh.max(axis=-1).reshape(-1, 1)
scores_fpfh = (scores_fpfh - min_scores_fpfh) / (max_scores_fpfh - min_scores_fpfh)
scores_fpfh = np.mean(scores_fpfh, axis=0)

**Comment by hzzzzzhappy** [comment]
Hi, thank you for your careful inspection.

In fact, this is a practical workaround. Since the original code was lost, we were not able to fully reproduce the exact performance of the original implementation. Due to limited time and maintenance resources, we adopted this strategy of leveraging different branches to score points and samples in order to achieve better empirical performance.

In other words, this is not a bug, but a technique used for fast reproduction of the results. We apologize for any confusion this may have caused.

Some recent works may provide you with useful inspiration. These approaches are generally easier to reproduce and maintain:

1. Recent works: https://github.com/[M-3LAB/UCA-3DL](https://github.com/M-3LAB/UCA-3DL)

2. Survey repository: https://github.com/M-3LAB/awesome-3d-anomaly-detection

**Comment by answer-miao** [comment]
Thank you for your reply. I have already adjusted the code. Thank you for your contribution to this field！
